contracts/ERC20RewardApeV1.sol

```diff
@@ -81,18 +84,23 @@ contract ERC20RewardApeV1 is ReentrancyGuard, Ownable, Initializable {
     event SkimStakeTokenFees(address indexed user, uint256 amount);
     event LogUpdatePool(uint256 bonusEndTime, uint256 rewardPerSecond);
     event EmergencyRewardWithdraw(address indexed user, uint256 amount);
-    event EmergencySweepWithdraw(address indexed user, IERC20 indexed token, uint256 amount);
+    event EmergencySweepWithdraw(
+        address indexed user,
+        IERC20 indexed token,
+        uint256 amount
+    );
 
     function initialize(
         IERC20 _stakeToken,
         IERC20 _rewardToken,
+        bool _isNativeTokenReward,
         uint256 _rewardPerSecond,
         uint256 _startTime,
         uint256 _bonusEndTime
-    ) external initializer
-    {
+    ) external initializer {
         STAKE_TOKEN = _stakeToken;
         REWARD_TOKEN = _rewardToken;
+        isNativeTokenReward = _isNativeTokenReward;
```

Comment:
Instead of passing the flag, check for `address(0)` so that `_rewardToken` and `_isNativeTokenReward` can never be out of sync.

https://github.com/ApeSwapFinance/apeswap-iao/blob/main/contracts/IAO.sol#L96